### PR TITLE
fix(tracing): Don't mask nationalids on http>4xx

### PR DESCRIPTION
**Pull request by sindrig**
Avoid masking antionalids in request urls when we don't get HTTP OK
Also add masking to express handlers as they were exposing national
ids accidentally.

This closes island-is/infrastructure#958

### CODE REVIEW

File: libs/infra-tracing/src/lib/datadog-tracer.ts [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
+import { Span } from 'opentracing'
 
 if (process.env.NODE_ENV !== 'development') {
+  const rewriteUrl = (url: string): string =>
+    url
+      .split('/')
+      .map((segm) => (isPerson(segm) ? '--MASKED--' : segm))
+      .join('/')
+  const shouldMask = (statusCode?: number): boolean =>
+    !statusCode || statusCode < 400
```
Comment (pshomov) [inline libs/infra-tracing/src/lib/datadog-tracer.ts:12]: We need an additional input here which is the magic env variable that will allow this to be possible at all. By default shouldMask should return `true` for all status codes.

File: libs/infra-tracing/src/lib/datadog-tracer.ts [hunk 2]
```diff
@@ -1,20 +1,41 @@
...
         request: (span, req, res) => {
           const url =
-            res?.url || `http(s)://${req?.getHeader('host')}${req?.path}`
-          const urlRewritten = url
-            .split('/')
-            .map((segm) => (isPerson(segm) ? '--MASKED--' : segm))
-            .join('/')
-          span?.setTag('http.url', urlRewritten)
+            res?.url ||
+            `${req?.protocol}//${req?.getHeader('host')}${req?.path}`
```
Comment (pshomov) [inline libs/infra-tracing/src/lib/datadog-tracer.ts:37]: I was under the impression req.protocol was not present very much, hence I forced it to be `http(s):`. Maybe have that as a fallback?

**Review by pshomov — COMMENTED** [review]
Needs a bit more functionality

File: libs/infra-tracing/src/lib/datadog-tracer.ts (hunk 2 above)
Comment (sindrig) [reply libs/infra-tracing/src/lib/datadog-tracer.ts:37]: It's a required attribute on `ClientRequest` and I have only seen it being present in my tests.

File: libs/infra-tracing/src/lib/datadog-tracer.ts (hunk 1 above)
Comment (sindrig) [reply libs/infra-tracing/src/lib/datadog-tracer.ts:12]: oh so this shouldn't be always on? talk in person

**Review by pshomov — APPROVED** [review]
LGTM

**Pull request merged**